Libplanet.Net/Consensus/RoundContext.cs

```diff
@@ -52,46 +57,69 @@ public class RoundContext<T>
 
         public BlockHash BlockHash { get; internal set; }
 
+        public PublicKey CurrentNodePublicKey => _validators[(int)NodeId];
+
         public long VoteCount
         {
             get
             {
                 lock (_lock)
                 {
-                    return _votes.Count;
+                    return _voteSet.Votes.Count(x => x.Flag == VoteFlag.Absent);
                 }
             }
         }
 
-        public long Vote(Address? address)
+        public long CommitCount
         {
-            Log.Debug(
-                "Vote({Before}/{Total}) NodeID: {Id}, Validator: {Address}, " +
-                "Height: {Height}, Round: {Round}",
-                _votes.Count,
-                _numberOfValidator,
-                NodeId,
-                address,
+            get
+            {
+                lock (_lock)
+                {
+                    return _voteSet.Votes.Count(x => x.Flag == VoteFlag.Commit);
+                }
+            }
+        }
+
+        public Vote Voting(VoteFlag flag)
+        {
+            return new Vote(
                 Height,
-                Round);
+                Round,
+                BlockHash,
+                DateTimeOffset.Now,
+                CurrentNodePublicKey,
+                flag,
+                NodeId,
+                Encoding.Default.GetBytes("sign").ToImmutableArray());
+        }
 
-            if (address is null)
+        public long Vote(Vote vote)
+        {
+            lock (_lock)
```

Comment:
This lock block is not required. How about move this logging code into the next lock block?